### PR TITLE
#162234049 Add recently published articles section

### PR DESCRIPTION
**Pull request by ucheg6**
#### What does this PR do?
Add recently published articles section
#### Description of Task to be completed?
- create and render recent Article component
- create reducer and action types
- create recent articles saga and dispatch action to fetch recently published articles

#### How should this be manually tested?
On command line ;
- git clone this repo
- npm install 
- Run `npm start:dev ` from the `haven-ah-frontend` directory

#### What are the relevant pivotal tracker stories?
[#162234049](https://www.pivotaltracker.com/story/show/1162234049)

#### Screenshots (if appropriate)>
screen shot 2018-12-03 at 4 56 28 pm

### CODE REVIEW

**Comment by wiztemple** [comment]
Good Job @ucheg6 .

File: src/components/containers/recentArticles/RecentArticles.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            <div className="trending-articles">
+              <h1 className="title is-2 mb-3">
+                Recently Published Articles
+              </h1>
+            </div>
+          </div>
+          < div className = "columns is-variable is-8 mt-3">
+            {
+              this.props.articles
+            && this.props.articles.slice(0, 3).map((article, index) => {
```
Comment (seyi-adeleke) [inline src/components/containers/recentArticles/RecentArticles.jsx]: Good job using Slice here. However you shouldn't  do this in the component. Either at the reducer or the store. Also shouldn't the api be returning the exact number of articles you need?

Comment (seyi-adeleke) [inline src/components/containers/recentArticles/RecentArticles.jsx]: Also its not a good idea to use the index as a key https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318

File: src/components/containers/recentArticles/RecentArticles.jsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+              return (<RecentArticleCard key={index} article={article} />);
+            })
+            }
+          </div>
+          < div className = "columns is-variable is-8 mt-3">
+            {
+              this.props.articles
+            && this.props.articles.slice(3, 6).map((article, index) => {
```
Comment (seyi-adeleke) [inline src/components/containers/recentArticles/RecentArticles.jsx]: 🤔

File: src/components/containers/recentArticles/recentArticles.css [hunk 3]
```diff
@@ -0,0 +1,68 @@
+@import url('https://fonts.googleapis.com/css?family=Playfair+Display');
+@import url('https://fonts.googleapis.com/css?family=Noto+Serif');
+@import url('https://fonts.googleapis.com/css?family=Sunflower:300');
+
+body {
+    padding: 0;
+    margin: 0;
+    /* font-family: 'Playfair Display', sans-serif; */
```
Comment (seyi-adeleke) [inline src/components/containers/recentArticles/recentArticles.css]: shouldn't the font be the same?

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]

File: src/components/containers/recentArticles/recentArticles.css (hunk 3 above)
Comment (seyi-adeleke) [reply src/components/containers/recentArticles/recentArticles.css]: Also your package lock shouldn't have a diff

Comment (ucheg6) [reply src/components/containers/recentArticles/recentArticles.css]: The fonts are the same. the font is coming from APP.scss.

File: src/components/containers/recentArticles/RecentArticles.jsx (hunk 1 above)
Comment (ucheg6) [reply src/components/containers/recentArticles/RecentArticles.jsx]: The get all articles api endpoint returns more than what I need. So I'm using the slice to display what I need

Comment (ucheg6) [reply src/components/containers/recentArticles/RecentArticles.jsx]: Yeah, the API returns the exact number of articles we need. The slice is just for displaying the cards. The Bulma style is expecting three cards in one row. If I don't control the display it displays 6 cards in one row.

Comment (ucheg6) [reply src/components/containers/recentArticles/RecentArticles.jsx]: Thanks, I've changed it to articleId.

File: src/components/containers/recentArticles/RecentArticles.jsx (hunk 2 above)
Comment (ucheg6) [reply src/components/containers/recentArticles/RecentArticles.jsx]: So I am doing the same thing here. Rendering the last three on a different row.

File: src/App.jsx [hunk 4]
```diff
@@ -3,10 +3,9 @@ import { BrowserRouter as Router } from 'react-router-dom';
 import { Provider } from 'react-redux';
 import Routes from './routes';
 import 'bulma/css/bulma.min.css';
-import 'font-awesome/css/font-awesome.min.css';
-
 import '../public/styles/App.scss';
 import '../public/styles/responsive.scss';
+import 'font-awesome/css/font-awesome.min.css';
```
Comment (seyi-adeleke) [inline src/App.jsx]: Same thing about absolute and relative imports

File: src/components/containers/recentArticles/RecentArticles.jsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  }
+}
+
+const mapStateToProps = state => ({
+  articles: state.recentArticles.articles,
+});
+
+RecentArticles.propTypes = {
+  recentArticlesAction: PropTypes.func.isRequired,
+  articles: PropTypes.array,
```
Comment (seyi-adeleke) [inline src/components/containers/recentArticles/RecentArticles.jsx:53]: add `.isRequired`

File: src/components/containers/recentArticles/recentArticlesCard.jsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+          </div>
+          <div className="ah-recentCard_bookmark"/>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+recentArticlesCard.propTypes = {
+  article: PropTypes.object,
```
Comment (seyi-adeleke) [inline src/components/containers/recentArticles/recentArticlesCard.jsx]: add proptypes for the other props you're using

**Review by TheoOkafor — COMMENTED** [review]
looks good.

**Review by seyi-adeleke — APPROVED** [review]

**Pull request merged**